docs/STYLE_GUIDE.md

```diff
@@ -156,3 +156,17 @@ duplicating documentation.
 - For larger modules, use one test file per extrinsic for unit tests. Make unit
   tests as decoupled as possible from other modules. Place end-to-end and
   integration tests in extra files.
+- If possible, test unreachable code and states thought to be impossible using
+  the following schema:
+
+```rust
+// In code logic
+zeitgeist_primitives::unreachable_non_terminating!(condition, log_target, message)
+```
+
+```rust
+// In test
+#[test]
+#[should_panic(expected = message)]
+// Cause assertion
+```
```

Comment:
````suggestion
  ```rust
  // In code logic
  zeitgeist_primitives::unreachable_non_terminating!(condition, log_target, message)
  ```
  
  ```rust
  // In test
  #[test]
  #[should_panic(expected = message)]
  // Cause assertion
  ```
````